pkg/simulate/pushsync/pushsync.go

```diff
@@ -0,0 +1,301 @@
+package pushsync
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/rand"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/beekeeper/pkg/bee"
+	"github.com/ethersphere/beekeeper/pkg/beeclient/api"
+	"github.com/ethersphere/beekeeper/pkg/beekeeper"
+	"github.com/ethersphere/beekeeper/pkg/random"
+	proxyClient "github.com/ethersphere/ethproxy/pkg/api/client"
+	"github.com/prometheus/client_golang/prometheus/push"
+)
+
+// Options represents simulation options
+type Options struct {
+	GasPrice         string
+	MetricsPusher    *push.Pusher
+	PostageAmount    int64
+	PostageDepth     uint64
+	PostageLabel     string
+	DownloadWait     time.Duration
+	UploadWait       time.Duration
+	DownloadCount    int64
+	DownloadRetry    int64
+	Seed             int64
+	ProxyApiEndpoint string
+	ChunkCount       int64
+	StartPercentage  float64
+	EndPercentage    float64
+	StepPercentage   float64
+}
+
+// NewDefaultOptions returns new default options
+func NewDefaultOptions() Options {
+	return Options{
+		GasPrice:         "",
+		MetricsPusher:    nil,
+		PostageAmount:    1000,
+		PostageDepth:     20,
+		PostageLabel:     "test-label",
+		UploadWait:       30 * time.Second,
+		DownloadWait:     15 * time.Second,
+		DownloadRetry:    3,
+		DownloadCount:    3,
+		Seed:             0,
+		ProxyApiEndpoint: "http://ethproxy.localhost",
+		ChunkCount:       500,
+		StartPercentage:  0.0,
```

Comment:
it might be worth a comment here that all three parameters are expected to be in the range of `0 <(=) v <(=) 1`, since one does not know if to put `60%` or `0.6` here